### PR TITLE
Deploy Sigstore Fulcio in the dev environment

### PR DESCRIPTION
**Pull request by zregvart**
This adds Helm to tools/go.mod so we can get updates as needed and configures Sigstore Fulcio deployment against Kubernetes OIDC provider.

To support Helm via tools/go.mod Docker dependencies needed to be downgraded.

This is first stab at getting Fulcio operational, while it does start and seems to be responsive there is more work in getting it fully usable.

Ref. https://issues.redhat.com/browse/HACBS-2082

### CODE REVIEW

File: hack/helm.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+# Copyright 2022 Red Hat, Inc.
```
Comment (lcarva) [inline hack/helm.sh:2]: Update copyright to this year?

File: hack/development/kustomization.yaml [hunk 2]
```diff
@@ -21,3 +21,4 @@ kind: Kustomization
 resources:
   - ../test
   - ../chains
+  - ../sigstore/fulcio
```
Comment (lcarva) [inline hack/development/kustomization.yaml:24]: Eventually, we want rekor and friends here as well, right?

Comment (zregvart) [reply hack/development/kustomization.yaml:24]: Is Rekor necessary for keyless?

File: tools/go.mod [hunk 3]
```diff
@@ -10,12 +10,19 @@ require (
...
 )
 
 // until https://github.com/grafeas/grafeas/pull/565 is merged/released
 replace github.com/grafeas/grafeas => github.com/zregvart/grafeas v0.0.0-20230102130750-341817cf5037
 
+// Helm v3 is using ORAS 1.2.2 which needs an older version of docker, this
+// helps with:
+//   packages/pkg/mod/oras.land/oras-go@v1.2.2/pkg/auth/docker/login_tls.go:181:30:
+//   assignment mismatch: 3 variables but registry.PingV2Registry returns 2 values
+replace github.com/docker/docker => github.com/docker/docker v20.10.24+incompatible
```
[the remark this answers is not in the document]
Comment (zregvart) [reply tools/go.mod:24]: I have no way of knowing, but I think most likely the code paths required for those will not be invoked

File: hack/development/kustomization.yaml (hunk 2 above)
Comment (lcarva) [reply hack/development/kustomization.yaml:26]: I _think_ it's possible to skip rekor, but it's definitely not something we want to do in a real environment. For instance, to disable rekor in cosign v2 you have to use the `--insecure-ignore-tlog` parameter. We can iterate here.

**Review by simonbaird — COMMENTED** [review]
Appreciate the multiple commits and explanations.

**Comment by zregvart** [comment]
I might have a look at disabling certain components, but this should deploy Sigstore services and the demo script shows getting certificate from Fulcio with signing and verifying of a blob.

File: hack/development/kustomization.yaml (hunk 2 above)
Comment (zregvart) [reply hack/development/kustomization.yaml]: Ended up using Rekor, need to see about ctlog and timestamp, perhaps in a subsequent PR...

File: hack/cosign-sign-blob-fulcio.sh [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# Installs Tekton Pipeline, the Enterprise Contract Policy custom resource and
```
Comment (lcarva) [inline hack/cosign-sign-blob-fulcio.sh:18]: Outdated description?

File: hack/ingress/kustomization.yaml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+---
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+nameSuffix: ""
+namespace: ingress-nginx
+
+resources:
+  - https://raw.githubusercontent.com/kubernetes/ingress-nginx/main/deploy/static/provider/kind/deploy.yaml
```
Comment (lcarva) [inline hack/ingress/kustomization.yaml:25]: Pin to a release/commit?

File: hack/setup-dev-environment.sh [hunk 6]
```diff
@@ -50,11 +50,22 @@ nodes:
...
+    nodeRegistration:
+      kubeletExtraArgs:
+        node-labels: "ingress-ready=true"
   extraPortMappings:
   - containerPort: ${REGISTRY_PORT:-5000}
     hostPort: ${REGISTRY_PORT:-5000}
     listenAddress: 127.0.0.1
     protocol: TCP
+  - containerPort: 80
+    hostPort: 80
```
Comment (lcarva) [inline hack/setup-dev-environment.sh:64]: It would be nice to not have to use a low port number here.

**Review by lcarva — APPROVED** [review]
I think this is a good starting point. My main concern is that there's a lot going on here that will be quite hard for anyone else in the team to debug. A bit of documentation on how to verify things would probably go a long way.

File: hack/setup-dev-environment.sh (hunk 6 above)
Comment (zregvart) [reply hack/setup-dev-environment.sh:64]: Yeah, perhaps in a followup, this would involve plugins, not sure if these are configurable via Helm charts...

**Review by lcarva — APPROVED** [review]

**Pull request merged**